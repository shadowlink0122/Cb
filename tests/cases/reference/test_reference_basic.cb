// 参照型の包括的テスト

void main() {
    // Test 1: 整数型の参照
    int a = 10;
    int& ref_a = a;
    println(ref_a);  // Expected: 10
    
    ref_a = 20;
    println(a);      // Expected: 20
    println(ref_a);  // Expected: 20
    
    // Test 2: 複数の参照
    int b = 30;
    int& ref_b = b;
    ref_b = 40;
    println(b);      // Expected: 40
    println(ref_b);  // Expected: 40
    
    // Test 3: 浮動小数点数の参照
    float f = 3.14;
    float& ref_f = f;
    println(ref_f);  // Expected: 3.140000
    
    ref_f = 2.71;
    println(f);      // Expected: 2.710000
    println(ref_f);  // Expected: 2.710000
    
    // Test 4: 参照から参照へ（連鎖）
    int x = 100;
    int& ref1 = x;
    int& ref2 = ref1;
    println(ref2);   // Expected: 100
    
    ref2 = 200;
    println(x);      // Expected: 200
    println(ref1);   // Expected: 200
    println(ref2);   // Expected: 200
}
