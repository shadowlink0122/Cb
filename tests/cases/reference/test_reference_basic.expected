10
20
20
40
40
3.140000
2.710000
2.710000
100
200
200
200
